Building a method that takes an array and returns a sorted array (using bubble sort methodology)

AhmedTohamy01 opened the pull request:
_**This project is about building two ruby methods that takes an array and returns a sorted array.**_ 
**_Both methods must use the bubble sort methodology ._**

_**First Method: #bubble_sort. : Sorting Numerical Array.**_

Problem Solving Steps:
1) Building a method that iterate over each item in the array and compare it to the next item(except the last item ) , if the first item is bigger than the next item both items will exchange places in the array. otherwise nothing will happen.

2) After iterate over all items in the array (except the last item), we should build another while loop to repeat the whole process again, to make sure that the array is properly sorted. This loop will run for a number of times equal to (n - 1) where n is the number of items in the array.

_**Second Method: #bubble_sort_by. : Sorting Strings Array.**_

Problem Solving Steps:
1) Building a method that uses "yielding methodology " to iterate over each item in the array and compare it to the next item(except the last item ) , if the length of first item is bigger than the length of the next item both items will exchange places in the array. otherwise nothing will happen.

2) After iterate over all items in the array (except the last item), we should build another while loop to repeat the whole process again, to make sure that the array is properly sorted. This loop will run for a number of times equal to (n - 1) where n is the number of items in the array.

pelzolga123 commented:
## Hi Team! ![](https://user-images.githubusercontent.com/38801423/62908122-703d4480-bd44-11e9-9115-42025ab7f341.png) 
### You did a great job, methods work without any errors :+1: 
### Kudos for amazingly descriptive Pull Request :1st_place_medal:  
Just a few changes are required  before this project can be approved:
* [ ] Please kindly rename `bubble_sort` method in bubble_sort_by.rb to `bubble_sort_by` so that it represents method`s functionality.
* [ ] No debugging instructions should be in committed code (as discussed [here](https://questions.microverse.org/t/debugging-instructions-code/346)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/section/debugging_requirement.png) from Pathwright) Last line in `bubble_sort.rb` file should be removed.
* [ ] You README file will look more professional if you add a brief description on how to clone and run your project using terminal commands.

### Please kindly fix the above issues and send another code review request.

If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

AhmedTohamy01 commented:
Hi Olga @pelzolga123 , Thank you so much for the comments .

Point 1: We did and renamed the method name.

Point 2: You refer us to a link we don't have access to is , and if we removed the last line in "bubble_sort" file as you suggested , lint is giving us an offense .

Point 3:  We didn't actually understand what do you mean by "clone and run the project using terminal commands".  i don't use any terminal commands to run my code!

pelzolga123 commented:
> Hi Olga @pelzolga123 , Thank you so much for the comments .
> 
> Kindly find our reply for your comments below.
> 
> **Point 1:** We did and renamed the method name.
> 
> **Point 2:** You refer us to a link we don't have access to it , try to open the link and you will see it will ask you for a username and password. and if we removed the last line in "bubble_sort" file as you suggested , lint is giving us an offense .
> 
> **Point 3:** We didn't actually understand what do you mean by "clone and run the project using terminal commands". i don't use any terminal commands to run my code! Why should i write an instructions to run my program using terminal commands ? anybody see the code can run it in his own favorite way.
> 
> This is Not a commercial project where there are a "unified" instructions to run the "program", it's just a "method".

1. Great job!

AhmedTohamy01 commented:
Hi Olga,

Thank you so much for taking time in clarifying what is Debugging Instructions which should be removed, we already removed the  "bubble_sort([5, 4, 3, 2, 1])" line as requested.
But i would like to mention again, that we still don't have access to the page that you refer us to it, our Microverse Accounts doesn't work there (me and my coding partner). 

For point 3: With total respect to your opinion. , We stick to the project requirements and We still see that this is NOT from the project requirements.

pelzolga123 commented:
> Hi Olga,
> 
> Thank you so much for taking time in clarifying what is Debugging Instructions which should be removed, we already removed the "bubble_sort([5, 4, 3, 2, 1])" line as requested.
> But i would like to mention again, that we still don't have access to the page that you refer us to it, our Microverse Accounts doesn't work there (me and my coding partner).
> 
> For point 3: With total respect to your opinion. , We stick to the project requirements and We still see that this is NOT from the project requirements.

That's great, I am happy to help. 

As for that website you can't access maybe I need to explain it more clearly. You need to type email and click the button `with email` on the right side. Then you will get a link on your email that will allow you to access all technical questions.

As for what I asked to do in the Readme is not just my own requirement. Microverse requirements say that the Readme file should look professional. When you will work on the Professional Skills section one of the tasks is to make a professional-looking Github Profile. And one of the requirements there will be: every Readme file should contain step-by-step installation guidelines and examples of usage. So I am just trying to help you make the Readme file look more professional. But if you don't wont to make that change it's up to you.

AhmedTohamy01 commented:
Hi Olga, @pelzolga123 

Thank you so much for your help.

For the Technical Question Website, it still didn't work after your advice , it asks for username and password, and didn't send any email to my email address (or junk mail).

For the ReadME file, absolutely we agree about that ReadMe file should looks professional all the time to make a professional-looking Github Profile. And we agree also that the "setup instructions" section should be there (IF Required), if we have a real program doing something and in order for the users to use it or install it they should be guided , then it will be required to add this section. But this is just a "Method" for educational purpose . It doesn't have even the GitFlow which is more essential than the ReadMe file make a professional-looking Github Profile. 

You got my point? 

**_But the most important point that we would like to Thank you really for your care about how our Github profile should looks like and your willingness to help , Much Appreciated._**

AhmedTohamy01 commented:
Hi Olga, @pelzolga123 

Can we ask for your help regarding the  "Debugging Instructions" Point,  could you please send us the content of the the link here in GitHub, so we can see exactly what is required from us ?

pelzolga123 commented:
> Hi Olga, @pelzolga123
> 
> Can we ask for your help regarding the "Debugging Instructions" Point, could you please send us the content of the the link here in GitHub, so we can see exactly what is required from us ?
I sent you a message on slack, let's move our conversation there.

Dandush03 commented:
# :heavy_check_mark: Project Approve :exploding_head: :checkered_flag:
### Hey :wave: 
### Great Job so Far :grey_exclamation: :grey_exclamation: :muscle: :smiley: 

### Following Previous Reviews and Seeing you Have Made Required Changes. 
### The project Looks Great.:grey_exclamation::wink:.

[__OPTIONAL__]

Keep your code in one file, There is no need to create one file per method :wink: this advice will be a most for your next project

### Good Luck in your Next Assignment
### Happy coding:computer: 
#### Daniel Laloush

AhmedTohamy01 commented:
Thank you Deniel, @Dandush03 

Much Appreciated :)
